src/sentry/api/serializers/models/alert_rule_trigger_action.py

```diff
@@ -39,7 +42,28 @@ def get_input_channel_id(self, action):
         """
         return action.target_identifier if action.type == action.Type.SLACK.value else None
 
-    def serialize(self, obj, attrs, user):
+    def get_attrs(self, item_list, user, **kwargs):
```

Comment:
In order to make the `SentryAppInstallation` lookup one call you'd need to do:
```py
def get_attrs(self, item_list, user, **kwargs):
    sentry_app_ids = {action.sentry_app_id for action in item_list}
    sentry_app_installations = SentryAppInstallation.objects.get_installed_for_organization(
        kwargs["organization_id"]
    ).filter(sentry_app_id_in=sentry_app_ids)
    sentry_app_installations_by_sentry_app_id = {
        installation.sentry_app_id: installation
        for installation in sentry_app_installations
    }

    result = defaultdict(dict)
    for action in item_list:
        sentry_app_installation = sentry_app_installations_by_sentry_app_id.get(action.sentry_app_id)
        if sentry_app_installation and action.sentry_app_config:
            component = sentry_app_installation.prepare_sentry_app_components(
                "alert-rule-action", None, action.sentry_app_config
            )
            result[action] = {"formFields": component.schema.get("settings", {})}

    return result
```
But each call to `prepare_sentry_app_components()` also does a DB lookup which is still bad.
You'll need to refactor that method to group those rows as well.

Reply:
👍  makes sense!